src/System.Net.Primitives/src/System/Net/IPAddress.cs

```diff
@@ -609,12 +610,19 @@ public override int GetHashCode()
         {
             // For IPv6 addresses, we cannot simply return the integer
             // representation as the hashcode. Instead, we calculate
-            // the hashcode from the string representation of the address.
+            // the hashcode from the address literals and ScopeId.
             if (IsIPv6)
             {
                 if (_hashCode == 0)
                 {
-                    _hashCode = StringComparer.OrdinalIgnoreCase.GetHashCode(ToString());
+                    int hashCode = HashHelpers.Combine(HashHelpers.RandomSeed, unchecked((int)PrivateScopeId));
+                    ushort[] numbers = _numbers;
```

Comment:
What is the purpose of the local? Does it give better codegen?

Reply:
Ah, I see your commit description. How do you know it produces better code -perf measurement?
@jkotas is that a known limitation, that we have to give that hint.

Reply:
.. continuing to read backwards, I see @justinvp 's comment :) I would measure the perf.

Reply:
Indeed seems to indeed be a bit better:

|                          Method |       Mean |     Error |    StdDev |
 |-------------------------------- |-----------:|----------:|----------:|
 |             GetHashCodeToString | 153.805 ns | 2.9882 ns | 3.0687 ns |
 |          GetHashCodeHashHelpers |   7.206 ns | 0.1791 ns | 0.1917 ns |
 | GetHashCodeHashHelpersWithLocal |   5.710 ns | 0.1489 ns | 0.2087 ns |

Reply:
Interesting. I guess it is a hint to enregister it.